Fix for opening a file with proxy custom node inside

marimano opened the pull request:
Base implementation of `DeserializeCore` uses ports of a node which are not added yet if the node is a proxy custom node. So this adding should be done before calling `base.DeserializeCore`.

I have got an error during opening a workspace which contains custom node instance with port info:
````
<Dynamo.Nodes.Function guid="8146efc7-9512-3bb3-c1ed-3242d9b65e07" type="Dynamo.Nodes.Function"
nickname="D" x="4644" y="4322" isVisible="true" isUpstreamVisible="true" lacing="Longest">
      <PortInfo index="0" default="True" />
      <PortInfo index="1" default="True" />
      <ID value="78a4e34c-93d8-d558-e47c-4ff9b1da1dac" />
      <Name value="D" />
      <Description value="" />
      <Inputs>
        <Input value="" />
        <Input value="" />
      </Inputs>
      <Outputs>
        <Output value="" />
      </Outputs>
    </Dynamo.Nodes.Function>
````

marimano commented on src/DynamoCore/Nodes/Custom Nodes/Function.cs:
```diff
@@ -219,6 +209,16 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
+
+            XmlNode descNode = childNodes.LastOrDefault(subNode => subNode.Name.Equals("Description"));
+            if (descNode != null && descNode.Attributes != null)
+                Description = descNode.Attributes["value"].Value;
```
just put `if-else` block before `base.DeserializeCore`

marimano commented:
Hi @pboyer, @Benglin, please take a look at this small fix

Benglin commented:
I have seen it, since `DeserializeCore` method is also used in undo-redo, copy-paste operations, can you try out and make sure those work well? I'm going to need @ke-yu to help looking at this as he's the expert in custom node.

ke-yu commented:
hi @marimano , seems .dyn file contains some legacy xml elements...could you please verify the scenario that 1. Open and update custom node (input, output) 2. Open a .dyn file which contains that custom node instance, to make sure that this custom node instance is always up-to-date?

marimano commented:
Hi @ke-yu,

My custom node `D` has 2 input and 1 output ports. I opened it from a file, removed 1 input, and then opened a `.dyn` which contains `D` instance - Dynamo threw an exception here:
https://github.com/DynamoDS/Dynamo/pull/3978/files#diff-1ddd248354fe37a47fa30c9410df12faL1467
So I added one more fix into this PR.

After that I tried out several scenarios:
1) open `D.dyf`, add 1 input, open `Home.dyn` - Home is loaded without any errors, the custom node instance corresponds to modified `D` (it has 3 inputs as in `D` tab)
2) open `D.dyf`, remove 1 input, open `Home.dyn` - Home is loaded without any errors, the custom node instance corresponds to modified `D` (it has 1 input as in `D` tab); save `D.dyf` with 1 input.
3) open `Home.dyn` - Home is loaded, the custom node instance is proxy and has 2 inputs;  open `D.dyf` - D is loaded correctly, the custom node instance in Home are updated (is not proxy anymore and has 1 input)

ke-yu commented:
Then LGTM.

Benglin commented:
Thanks @marimano, and of course @ke-yu for helping out with the review!
